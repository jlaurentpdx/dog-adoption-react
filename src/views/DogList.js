import { useEffect, useState } from 'react';
import { fetchDogs } from '../services/dogs';

export default function DogList() {
  const [loading, setLoading] = useState(true);
  const [dogs, setDogs] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogs();
      setDogs(data);
      setLoading(false);
    };
    fetchData();
  }, []);

  if (loading) return <h1>Please wait...</h1>;

  return (
    <>
      {dogs.map((dog) => (
        <div key={dog.id} clasName="dog-card">
          <h2>{dog.name}</h2>
          <img src={dog.image} />
          <p>
            Meet {dog.name}, the {dog.breed}. {dog.bio}
          </p>
        </div>
      ))}
      ;
    </>
  );
}
