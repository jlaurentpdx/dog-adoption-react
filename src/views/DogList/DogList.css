.dog-list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 1%;
  align-items: top;
  justify-content: center;
}

.dog-link {
  text-decoration: none;
  color: var(--darker);
  border-radius: 10px;
  margin: 1%;
  width: 30vw;
  min-width: 400px;
  max-width: 500px;
  line-height: 0.5rem;
}

.dog-link:hover {
  color: var(--darker);
  background-color: var(--light);
}

.loader {
  font-family: 'Indie Flower', sans-serif;
  color: var(--darker);
}

h1 {
  font-family: 'Indie Flower', sans-serif;
  text-decoration: none;
}

img {
  border-radius: 10px;
  width: 30vw;
  min-width: 250px;
  max-width: 350px;
  border: 10px double var(--lighter);
}

p {
  font-family: 'Indie Flower', sans-serif;
  font-weight: bold;
  color: var(--dark);
}
