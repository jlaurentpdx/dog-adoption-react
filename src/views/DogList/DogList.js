import { useEffect, useState } from 'react';
import { fetchDogs } from '../../services/dogs';
import DisplayList from '../../components/DisplayList/DisplayList';

import './DogList.css';

export default function DogList() {
  const [loading, setLoading] = useState(true);
  const [dogs, setDogs] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogs();
      setDogs(data);
      setLoading(false);
    };
    fetchData();
  }, []);

  if (loading) return <h1 className="loader">...checking our list of furry friends...</h1>;

  return <DisplayList dogs={dogs} />;
}
