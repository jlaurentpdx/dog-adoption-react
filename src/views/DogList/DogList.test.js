import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import DogList from './DogList';

test('DogList renders a list of dogs', async () => {
  const { container } = render(
    <MemoryRouter>
      <DogList />
    </MemoryRouter>
  );

  screen.getByText('...checking our list of furry friends...');

  await screen.findByText('Barton');
  await screen.findByText('Liliana');
  await screen.findByText('Melba');

  expect(container).toMatchSnapshot();
});
