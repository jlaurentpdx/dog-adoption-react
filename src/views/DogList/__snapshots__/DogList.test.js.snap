// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogList renders a list of dogs 1`] = `
<div>
  <div
    class="dog-list"
  >
    <a
      class="dog-link"
      href="/dogs/1"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Barkley
        </h1>
        <img
          src="https://placedog.net/400/400/1"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/2"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Goldie
        </h1>
        <img
          src="https://placedog.net/400/400/2"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/3"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Beatrix
        </h1>
        <img
          src="https://placedog.net/400/400/3"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/4"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Jubilee
        </h1>
        <img
          src="https://placedog.net/400/400/4"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/5"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Luna
        </h1>
        <img
          src="https://placedog.net/400/400/5 "
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/6"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Sadie
        </h1>
        <img
          src="https://placedog.net/400/400/6"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/7"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Perry
        </h1>
        <img
          src="https://placedog.net/400/400/7"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/8"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Urso
        </h1>
        <img
          src="https://placedog.net/400/400/8"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/9"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Milo
        </h1>
        <img
          src="https://placedog.net/400/400/9"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/10"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Dutch
        </h1>
        <img
          src="https://placedog.net/400/400/10"
        />
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/11"
    >
      <div>
        <p>
          meet
        </p>
        <h1>
          Butters
        </h1>
        <img
          src="https://placedog.net/400/400/11"
        />
      </div>
    </a>
  </div>
</div>
`;
